plc/VersionInfo2.c

```diff
@@ -126,6 +129,13 @@ signed VersionInfo2 (struct plc * plc)
 
 #endif
 
+#if 0
+		/* firmware doesn't fill these fields properly, so don't check */
```

Comment:
Instead of adding more of these ugly #if 0, i suggest only a comment which clarify that the cookie value here isn't reliable.